src/comparator.rs

```diff
@@ -1,30 +1,482 @@
+
+#[derive(Copy, Clone, Eq, PartialEq)]
+pub enum RefintInput {
+    /// VRefint * 1/4
+    VRefintM14 = 0b0000,
+    /// VRefint * 1/2
+    VRefintM12 = 0b0001,
+    /// VRefint * 3/4
+    VRefintM34 = 0b0010,
+    /// VRefint
+    VRefint = 0b0011,
+}
+
+macro_rules! refint_input {
+    ($COMP:ident) => {
+        impl NegativeInput<$COMP> for RefintInput {
+            fn setup(&self, comp: &$COMP) {
+                comp.csr()
+                    .modify(|_, w| unsafe { w.inmsel().bits(*self as u8) })
+            }
+        }
+    };
+}
+
+refint_input!(COMP1);
+refint_input!(COMP2);
+
+macro_rules! dac_input {
+    ($COMP:ident, $channel:ty, $bits:expr) => {
+        impl NegativeInput<$COMP> for &$channel {
+            fn setup(&self, comp: &$COMP) {
+                comp.csr().modify(|_, w| unsafe { w.inmsel().bits($bits) })
+            }
+        }
+    };
+}
+
+#[cfg(any(feature = "stm32g071", feature = "stm32g081"))]
+dac_input!(COMP1, dac::Channel1<dac::Enabled>, 0b0100);
+#[cfg(any(feature = "stm32g071", feature = "stm32g081"))]
+dac_input!(COMP1, dac::Channel2<dac::Enabled>, 0b0101);
+
+#[cfg(any(feature = "stm32g071", feature = "stm32g081"))]
+dac_input!(COMP2, dac::Channel1<dac::Enabled>, 0b0100);
+#[cfg(any(feature = "stm32g071", feature = "stm32g081"))]
+dac_input!(COMP2, dac::Channel2<dac::Enabled>, 0b0101);
+
+pub struct Comparator<C> {
+    regs: C,
+}
+
+pub trait ComparatorExt<COMP> {
+    fn init<P: PositiveInput<COMP>, N: NegativeInput<COMP>>(
+        &mut self,
+        positive_input: P,
+        negative_input: N,
+        config: Config,
+    );
+    fn output(&self) -> bool;
+    fn enable(&self);
+    fn disable(&self);
+    fn output_pin<P: OutputPin<COMP>>(&self, pin: P);
+    fn listen(&self, edge: SignalEdge, exti: &EXTI);
+    fn unlisten(&self, exti: &EXTI);
+    fn is_pending(&self, edge: SignalEdge, exti: &EXTI) -> bool;
+    fn unpend(&self, exti: &EXTI);
+}
+
+macro_rules! comparator_ext {
+    ($COMP:ty, $Comparator:ty, $Event:expr) => {
+        impl ComparatorExt<$COMP> for $Comparator {
+            // TODO: Require init before any other functions?
+            fn init<P: PositiveInput<$COMP>, N: NegativeInput<$COMP>>(
+                &mut self,
+                positive_input: P,
+                negative_input: N,
+                config: Config,
+            ) {
+                positive_input.setup(&self.regs);
+                negative_input.setup(&self.regs);
+                self.regs.csr().modify(|_, w| unsafe {
+                    w.hyst()
+                        .bits(config.hysteresis as u8)
+                        .polarity()
+                        .bit(config.inverted)
+                        .pwrmode()
+                        .bits(config.power_mode as u8)
+                        .winout()
+                        .bit(config.output_xor)
+                });
+                // TODO: Delay for comp scaler bridge voltage stabilization?
```

Comment:
Hmm, no sure if we have a proper way for adding delay here. Maybe it should be just documented.
@dotcypress what do you think?

Reply:
I added a delay implementation that uses core_clk to calculate a busy-loop delay. It's only 200us but it could perhaps just be documented. I haven't tested but I don't think the peripheral will break if the delay isn't done, the negative input will just have an incorrect value initially.